proposed/2022/cpm-dotnet-cli-add-package-support.md

```diff
@@ -0,0 +1,134 @@
+
+    <ItemGroup>
+    </ItemGroup>
+</Project>
+```
+
+#### After `add package` is executed:<br>
+
+The props file: <br>
+```xml
+<Project>
+    <PropertyGroup>
+        <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    </PropertyGroup>
+    <ItemGroup>
+    <PackageVersion Include="Newtonsoft.Json" Version="13.0.1"/>
+    </ItemGroup>
+</Project>
+```
+The .csproj file: <br>
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
+        <TargetFramework>net6.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include="Newtonsoft.Json"/>
+    </ItemGroup>
+</Project>
+```
+
+In case there are multiple `Directory.packages.props` files in the repo, the props file that is closest must be considered.
+
+```xml
+Repository
+ |-- Directory.Packages.props
+ |-- Solution1
+    |-- Directory.Packages.props
+    |-- Project1
+ |-- Solution2
+    |-- Project2
+```
+
+In the above example, the following scenarios are possible:
+1. Project1 will evaluate the Directory.Packages.props file in the Repository\Solution1\ directory.
+2. Project2 will evaluate the Directory.Packages.props file in the Repository\ directory.
+
+***Sourced from https://devblogs.microsoft.com/nuget/introducing-central-package-management/
+
+### 2. The package reference does exist
+If the package already exists in `Directory.packages.props` the version should be updated in `Directory.packages.props`. The package version should either be the latest version or the one specified in the CLI command. The `<PackageReference>` in the project file should not change.
+
+#### Before `add package` is executed:<br>
+```xml
+<Project>
+    <PropertyGroup>
+        <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    </PropertyGroup>
+    <ItemGroup>
+    <PackageVersion Include="Newtonsoft.Json" Version="12.0.1"/>
+    </ItemGroup>
+</Project>
+```
+
+#### After `add package` is executed:<br>
+```xml
+<Project>
+    <PropertyGroup>
+        <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    </PropertyGroup>
+    <ItemGroup>
+    <PackageVersion Include="Newtonsoft.Json" Version="13.0.1"/>
+    </ItemGroup>
+</Project>
+```
+
+In case there are multiple `Directory.packages.props` files in the repo, all of them should be searched for a matching package reference. Once a matching package reference is found, its package version can be updated appropriately. 
+
+### Other Scenarios that can be considered
+
+While the above cases must be considered to solve the issue, there are a few other cases that can be considered to make the product more usable:
+
+1. If the project has the `<ManagePackageVersionsCentrally>` set to true, but the `Directory.packages.props` file has not been created yet the file should be created. At this point, package references can be updated according to the cases discussed above.
```

Comment:
It's worth calling out what would actually happen here today, even if we're not promising/committing to  the improvements themselves.

Reply:
I added a table in the revised proposal https://github.com/NuGet/Home/pull/11873 explaining current and new behavior for every scenario.